cmd/influx_tsm/converter.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/influxdb/influxdb/tsdb/engine/tsm1"
+)
+
+type KeyIterator interface {
+	Next() bool
+	Read() (string, []tsm1.Value, error)
+}
+
+// Converter encapsulates the logic for converting b*1 shards to tsm1 shards.
+type Converter struct {
+	path           string
+	maxTSMFileSize uint32
+	generation     int
+}
+
+// NewConverter returns a new instance of the Converter.
+func NewConverter(path string, sz uint32) *Converter {
+	return &Converter{
+		path:           path,
+		maxTSMFileSize: sz,
+	}
+}
+
+// Process writes the data provided by iter to a tsm1 shard.
+func (c *Converter) Process(iter KeyIterator) error {
+	// Ensure the tsm1 directory exists.
+	if err := os.MkdirAll(c.path, 0777); err != nil {
+		return err
+	}
+
+	w, err := c.nextTSMWriter()
+	if err != nil {
+		return err
+	}
+	defer w.Close()
```

Comment:
Won't this miss a call to `Close` if the file rolls over on L61?  It may also double close the file.